src/pages/HomePage/CreateAppModal.js

```diff
@@ -0,0 +1,203 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import {
+ Row, Col, Icon, Button, Modal, Input, Radio, Menu, Dropdown,
+} from 'antd';
+
+import {
+ label, input, radiobtn,
+} from './styles';
+
+import { createApp } from '../../actions';
+
+const RadioGroup = Radio.Group;
+
+class CreateAppModal extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      appName: '',
+      plan: 'free',
+      hasJSON: false,
+      category: 'generic',
+      elasticVersion: '5',
+    };
+  }
+
+	handleOk = async () => {
+		const { appName, category, elasticVersion } = this.state;
+		const { handleCreateApp } = this.props;
+		const options = {
+			appName,
+			category,
+			es_version: elasticVersion,
+		};
+
+		handleCreateApp(options);
+	};
+
+	handleChange = (e) => {
+		const { target: { name, value } } = e;
+		this.setState({
+			[name]: value,
+		});
+	};
+
+	handleCancel = () => {
+		this.props.handleModal();
+	};
+
+	handleMenuClick = (e) => {
+		const category = e.key;
+		this.setState({
+			category,
+		});
+	};
+
+  componentWillReceiveProps = (nextProps) => {
+    const { createdApp } = nextProps;
+    const { hasJSON, appName } = this.state;
+    if (createdApp.data && createdApp.data.id) {
+      if (hasJSON) {
+        window.location.href = `app/${appName}/import`;
```

Comment:
Please don't use `window.location` to change routes. Push to `props.history` instead.